### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by AlvaroDazaQuiroga**

### CODE REVIEW

**Comment by AlvaroDazaQuiroga** [comment]
@carledriss  ready for code review

File: src/main/java/org/fundacionjala/CodeWarsKata/EANValidator.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+     *
+     * @param code String , the code recived
+     * @return shows the result , true or false
+     */
+    public static boolean validate(String code) {
+        int lastValue = Character.getNumericValue(code.charAt(code.length() - PENULT_VENTOR_POSITION));
+        int[] checkSum = fillTheVector(code);
+        int total = 0;
+        for (int i = FOR_INIT; i <= code.length() - LENGTH_ARRAY_LIMIT; i++) {
+            if (i % PAIR_ODD_DIVIDER == 0) {
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/CodeWarsKata/EANValidator.java]: you can improve this code.

``` java
total += i % PAIR_ODD_DIVIDER == 0 ? checkSum[i] : checkSum[i] * ODD_MULTIPLICATOR_VALUE;
```

File: src/main/java/org/fundacionjala/CodeWarsKata/EANValidator.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    /**
+     * calculate the boolean value
+     *
+     * @param lastValue integer the las value of the vector
+     * @param total     integer, the result of the add of pair and add positions
+     * @return return true or fale
+     */
+
+    private static boolean checkSumResult(int lastValue, int total) {
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/CodeWarsKata/EANValidator.java]: Several lines of code. :(
You can try this.

``` java
int checkSumValue = total % CHECKSUM_FORMULA_CONDITION == RESULT_CONDITION ? 0 : VALUE_OF_THE_FORMULA - (total % VALUE_OF_THE_FORMULA);
return checkSumValue == lastValue;
```

File: src/main/java/org/fundacionjala/CodeWarsKata/EANValidator.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+     * calculate the total value of the pair and odd number
+     *
+     * @param code String , the code recived
+     * @return shows the result , true or false
+     */
+    public static boolean validate(String code) {
+        int lastValue = Character.getNumericValue(code.charAt(code.length() - PENULT_VENTOR_POSITION));
+        int[] checkSum = fillTheVector(code);
+        int total = 0;
+        for (int i = FOR_INIT; i <= code.length() - LENGTH_ARRAY_LIMIT; i++) {
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/CodeWarsKata/EANValidator.java]: It is not necessary to have a constant for FOR_INIT.
Remove it!.

File: src/main/java/org/fundacionjala/CodeWarsKata/EANValidator.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    public static final int CHECKSUM_FORMULA_CONDITION = 10;
+    public static final int VALUE_OF_THE_FORMULA = 10;
+    public static final int FOR_INIT = 0;
+    public static final int LENGTH_ARRAY_LIMIT = 2;
+    public static final int ODD_MULTIPLICATOR_VALUE = 3;
+    public static final int VECTOR_LENGTH_INIT = 13;
+    public static final int PAIR_ODD_DIVIDER = 2;
+    public static final int RESULT_CONDITION = 0;
+    public static final int PENULT_VENTOR_POSITION = 1;
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/CodeWarsKata/EANValidator.java]: These constants should be privates

**Comment by carledriss** [comment]
@AlvaroDazaQuiroga 
Let me know when you have fixed the observations.

**Comment by AlvaroDazaQuiroga** [comment]
@carledriss ready for code review

File: src/main/java/org/fundacionjala/CodeWarsKata/EANValidator.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+     *
+     * @param lastValue integer the las value of the vector
+     * @param total     integer, the result of the add of pair and add positions
+     * @return return true or fale
+     */
+
+    private static boolean checkSumResult(int lastValue, int total) {
+        int checkSumValue = total % CHECKSUM_FORMULA_CONDITION == RESULT_CONDITION ? 0 :
+                VALUE_OF_THE_FORMULA - (total % VALUE_OF_THE_FORMULA);
+        return checkSumValue == lastValue ? true : false;
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/CodeWarsKata/EANValidator.java]: Improve this line.

File: src/main/java/org/fundacionjala/CodeWarsKata/EANValidator.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+public class EANValidator {
+
+    private static final int CHECKSUM_FORMULA_CONDITION = 10;
+    private static final int VALUE_OF_THE_FORMULA = 10;
+    private static final int LENGTH_ARRAY_LIMIT = 2;
+    private static final int ODD_MULTIPLICATOR_VALUE = 3;
+    private static final int VECTOR_LENGTH_INIT = 13;
+    private static final int PAIR_ODD_DIVIDER = 2;
+    private static final int RESULT_CONDITION = 0;
+    private static final int PENULT_VENTOR_POSITION = 1;
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/CodeWarsKata/EANValidator.java]: remove this unnecessary variable

**Comment by AlvaroDazaQuiroga** [comment]
@carledriss ready for code review

**Comment by carledriss** [comment]
Looks good to me!

**Pull request merged**